buildpack.md

```diff
@@ -476,35 +476,62 @@ The purpose of launch is to modify the running app environment using app-provide
 **GIVEN:**
 - An OCI image exported by the lifecycle,
 - An optional process type specified by `CNB_PROCESS_TYPE`, and
-- Bash version 3 or greater,
-
-When the OCI image is launched,
+- Bash version 3 or greater, if needed,
 
-1. The lifecycle MUST source each file in each `<layers>/<layer>/profile.d` directory using the same Bash shell process,
-   1. Firstly, in order of `/bin/build` execution used to construct the OCI image.
-   2. Secondly, in alphabetically ascending order by layer directory name.
-   3. Thirdly, in alphabetically ascending order by file name.
+First, the lifecycle MUST locate a start command and choose an execution strategy.
 
-2. In the Bash shell process used to source the `profile.d` scripts, the lifecycle MUST source `<app>/.profile` if it is present.
+To locate a start command,
 
-3. If `CMD` in the container configuration is not empty, the lifecycle MUST join each argument with a space and execute the resulting command in the container using the Bash shell process used to source the `profile.d` scripts.
+1. **IF** `CMD` in the container configuration is not empty,
+    **THEN** the value of `CMD` is chosen as the start command.
 
-4. If `CMD` in the container configuration is empty,
+2. **IF** `CMD` in the container configuration is empty,
    1. **IF** the `CNB_PROCESS_TYPE` environment variable is set,
       1. **IF** the value of `CNB_PROCESS_TYPE` corresponds to a process in `<layers>/launch.toml`, \
-         **THEN** the lifecycle MUST execute the corresponding command in the container using the Bash shell process used to source the `profile.d` scripts.
+         **THEN** the lifecycle MUST choose the corresponding process as the start command.
 
       2. **IF** the value of `CNB_PROCESS_TYPE` does not correspond to a process in `<layers>/launch.toml`, \
          **THEN** launch fails.
 
    2. **IF** the `CNB_PROCESS_TYPE` environment variable is not set,
       1. **IF** there is a process with a `web` process type in `<layers>/launch.toml`, \
-         **THEN** the lifecycle MUST execute the corresponding command in the container using the Bash shell process used to source the `profile.d` scripts.
+         **THEN** the lifecycle MUST choose the corresponding process as the start command.
 
       2. **IF** there is not a process with a `web` process type in `<layers>/launch.toml`, \
          **THEN** launch fails.
+         
+To choose an execution strategy,
+
+1. **IF** the value of `CMD` is chosen as the start command,
+   1. **IF** the first parameter of `CMD` is not `--`,
+      **THEN** the lifecycle MUST invoke the value as a command using Bash with subsequent entries as arguments.
+
+   2. **IF** the first parameter of `CMD` is `--` and the length of `CMD` is greater than one,
```

Comment:
Curious, is `--` an existing standard or something we're inventing here so we know when to switch?

Reply:
As discussed during wg meeting, `--` convention from coreutils, git, bash, etc.